File: Typescript/985.ts
function isEven(num: number) {
  return num % 2 === 0;
}

function sumEvenAfterQueries(nums: number[], queries: number[][]): number[] {
  const answer: number[] = [];

  let sumEven = nums.reduce(
    (prev, next) => (isEven(next) ? prev + next : prev),
    0
  );

  for (const [value, index] of queries) {
    if (isEven(nums[index])) {
      sumEven -= nums[index];
    }

    nums[index] = nums[index] + value;

    if (isEven(nums[index])) {
      sumEven += nums[index];
    }

    answer.push(sumEven);
  }

  console.log(answer);

  return answer;
}

File: C/985.c


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *sumEvenAfterQueries(int *nums, int numsSize, int **queries, int queriesSize, int *queriesColSize, int *returnSize)
{
    int sumEven = 0;

    int *answer = (int *)malloc(numsSize * sizeof(int));

    *returnSize = numsSize;

    for (int i = 0; i < numsSize; i++)
    {
        if (nums[i] % 2 == 0)
        {
            sumEven += nums[i];
        }
    }

    for (int i = 0; i < queriesSize; i++)
    {
        int value = queries[i][0];
        int index = queries[i][1];

        if (nums[index] % 2 == 0)
        {
            sumEven -= nums[index];
        }

        nums[index] = nums[index] + value;

        if (nums[index] % 2 == 0)
        {
            sumEven += nums[index];
        }

        answer[i] = sumEven;
    }

    return answer;
}